apps/fyler/src/fyler_category_manager.erl

```diff
@@ -0,0 +1,672 @@
+-module(fyler_category_manager).
+-behaviour(gen_fsm).
+
+-include("../include/log.hrl").
+-include("fyler.hrl").
+
+-define(Regex, "\"PrivateIpAddress\": \"(?<ip>[^\"]*)\"").
+-define(Node(Category, Ip), list_to_atom(lists:flatten(io_lib:format("fyler_~p_pool@~s", [Category, Ip])))).
+-define(Timeout, 60000).
+
+%% ------------------------------------------------------------------
+%% API Function Exports
+%% ------------------------------------------------------------------
+
+-export([
+  start_link/1,
+  run_task/2,
+  cancel_task/2,
+  task_accepted/3,
+  task_rejected/3,
+  pool_connected/2,
+  pool_down/2,
+  pool_enabled/2,
+  pool_disabled/2
+]).
+
+%% ------------------------------------------------------------------
+%% gen_fsm Function Exports
+%% ------------------------------------------------------------------
+
+-export([init/1, handle_event/3,
+         handle_sync_event/4, handle_info/3,
+         terminate/3, code_change/4]).
+
+-export([start/2, zero/2, tasks/2, pools/2]).
+
+-record(state, {
+  category :: atom(),
+  retasks = fyler_queue:new() :: fyler_queue:fyler_queue(task()),
+  tasks = fyler_queue:new() :: fyler_queue:fyler_queue(task()),
+  task_filter = [] :: [task()],
+  pools = [] :: [{atom(), enabled | disabled | {pending, reference(), non_neg_integer()}}],
```

Comment:
Line 42 is too long:   pools = [] :: [{atom(), enabled | disabled | {pending, reference(), non_neg_integer()}}],.